tests/metric_providers/test_electricitymaps_provider.py

```
@@ -0,0 +1,428 @@
+    with patch('requests.get', return_value=make_response(records)):
+        df = provider._read_metrics()
+
+    # 3-second window at 1 s steps → 4 rows (0 s, 1 s, 2 s, 3 s)
+    assert len(df) == 4
+    assert (df['value'] == 42).all()
+    assert df['time'].is_monotonic_increasing
+
+
+# --- Live integration test (hits the real Electricity Maps API) ---
+
+LIVE_TOKEN = 'WgKazgtDkyxaMHAvSFYY'
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to: "LIVE_TOKEN"

(S105)